### PR TITLE
Move `elastic/apm-server` documentation source files

### PR DESCRIPTION
**Pull request by bmorelli25**
### Summary

This PR cherry-picks every commit made to the `docs` dir in the [elastic/apm-server](https://github.com/elastic/apm-server) repository.

### CODE REVIEW

File: docs/en/apm-server/release-notes.asciidoc [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+* {apm-php-ref-v}/release-notes.html[PHP agent]
+* {apm-py-ref-v}/release-notes.html[Python agent]
+* {apm-rum-ref-v}/release-notes.html[Real User Monitoring (RUM) agent]
+* {apm-ruby-ref-v}/release-notes.html[Ruby agent]
+
+**APM extensions**
+
+* https://github.com/elastic/apm-aws-lambda/blob/main/CHANGELOG.asciidoc[Elastic APM AWS Lambda extension]
+
+include::{root-dir}/CHANGELOG.asciidoc[]
```
Comment (colleenmcginnis) [inline docs/en/apm-server/release-notes.asciidoc:48]: Should we update this here?

```suggestion
include::{apm-server-root}/CHANGELOG.asciidoc[]
```

Comment (bmorelli25) [reply docs/en/apm-server/release-notes.asciidoc:48]: Doh! Yes

**Comment by bmorelli25** [comment]
I'm going to enable merge commits on this repo temporarily. After merging this PR with a merge commit, I'll disable them again.

**Review by colleenmcginnis — APPROVED** [review]
✨  WOOHOO! ✨ 

(make sure to document what finally worked!)

**Pull request merged**